examples/ds18b20/main.c

```diff
@@ -63,6 +63,13 @@ int main()
                                                         devices[i].id,
                                                         temperature);
 
+            if (temperature_p == NULL) {
+                strncpy(temperature, FSTR("failed to get"),
+                        sizeof(temperature));
+                temperature[sizeof(temperature) - 1] = '\0';
+                temperature_p = temperature;
+            }
+
```

Comment:
I think making std_print* to cleanly handle NULL is a good idea (I can have a look at that later - I have no idea how involved that is), but personally I still like to have the error check there to remind people they should check for errors and decide how to handle it.

As for `temperature_p = "failed to get";` - the reason I didn't do that is because the environment I used to code in (FreeBSD) uses -Wwrite-strings by default, which means literal strings are declared const char*, so combined with warnings about const, `char *p = "foo";` would be a compiler error.

In this case the string clearly safe, so I'm just going to change it. I might then later have a look at feasibility of adding `const` to the prototypes etc.

Thanks!